File: plateforme/src/main/java/org/steam2/plateforme/entites/GameType.java
package org.steam2.plateforme.entites;

/*
Codé par Jules
 */
public enum GameType {
    ACTION, ADVENTURE, FIGHTING, HORROR, MMORPG, MOBA, PLATEFORMER, PUZZLE,
    RACING, RPGS, SANDBOX, SIMULATION, SHOOTER, SPORTS, SURVIVAL;

    @Override
    public String toString() {
        switch (this) {
            case ACTION:
                return "ACTION";
            case ADVENTURE:
                return "ADVENTURE";
            case FIGHTING:
                return "FIGHTING";
            case HORROR:
                return "HORROR";
            case MMORPG:
                return "MMORPG";
            case MOBA:
                return "MOBA";
            case PLATEFORMER:
                return "PLATEFORMER";
            case PUZZLE:
                return "PUZZLE";
            case RACING:
                return "RACING";
            case RPGS:
                return "RPGS";
            case SANDBOX:
                return "SANDBOX";
            case SIMULATION:
                return "SIMULATION";
            case SHOOTER:
                return "SHOOTER";
            case SURVIVAL:
                return "SURVIVAL";
            case SPORTS:
                return "SPORTS";
        }
        return "";
    }
}

File: plateforme/src/main/kotlin/plateforme/entites/type/GameType.kt
package org.steam2.plateforme.plateforme.entites.type

/**
 * Enumération des types de jeux
 * @author Jules
 */
enum class GameType(val libelle: String) {
    ACTION("Action"),
    ADVENTURE("Adventure"),
    FIGHTING("Fighting"),
    HORROR("Horror"),
    MMORPG("MorpG"),
    MOBA("Moba"),
    PLATEFORMER("Plateformer"),
    PUZZLE("Puzzle"),
    RACING("Racing"),
    RPGS("Rpgs"),
    SANDBOX("Sandbox"),
    SIMULATION("Simulation"),
    SHOOTER("Shooter"),
    SPORTS("Sports"),
    SURVIVAL("Survival");

    override fun toString(): String {
        return libelle
    }
}